File: backend/src/main/java/app/miniappspring/controller/UserController.java
package app.miniappspring.controller;

import app.miniappspring.arguments.UpdateDataUserArgument;
import app.miniappspring.dto.delivey.UpdateDeliveryDataUser;
import app.miniappspring.dto.user.UpdateAvatarUserDto;
import app.miniappspring.dto.user.UpdateDataUserDto;
import app.miniappspring.entity.Role;
import app.miniappspring.service.AuthenticationService;
import app.miniappspring.service.UserService;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.Set;


@RestController
@RequestMapping("/api/v1/user")
@RequiredArgsConstructor
public class UserController {
    private final AuthenticationService authenticationService;
    private final UserService userService;

    @PreAuthorize("hasAnyAuthority('ROLE_USER')")
    @GetMapping
    public ResponseEntity<String> sayHello() {
        return ResponseEntity.ok("Hi " + authenticationService.getAuthenticationInfo().getUsername());
    }

    //    http://localhost:8080/api/v1/user/remove/nik3
    @GetMapping("/remove/{username}")
    public ResponseEntity<String> sayHello(@PathVariable @NonNull String username) {
        return ResponseEntity.ok("Удаление пользователя прошло:  " + userService.remove(username));
    }

    @PostMapping("/data/update")
    public UpdateDataUserDto updateDataUser(@RequestBody UpdateDataUserArgument updateDataUserArgument) {
        return userService.updateDataUser(updateDataUserArgument);
    }

    @PostMapping("/data/update/delivery")
    public UpdateDeliveryDataUser updateDataUserAboutDelivery(@RequestBody UpdateDeliveryDataUser updateDeliveryDataUser) {
        return userService.updateDataUserAboutDelivery(updateDeliveryDataUser);
    }

    @GetMapping("/data/delivery")
    public UpdateDeliveryDataUser getDataUserAboutDelivery() {
        return userService.getDataUserAboutDelivery();
    }

    @PostMapping("/photo/update")
    public ResponseEntity<Boolean> updatePhotoUser(@RequestParam String username, @RequestParam MultipartFile multipartFileAvatar) {
        return ResponseEntity.ok(userService.updateUserAvatar(new UpdateAvatarUserDto(username, multipartFileAvatar)));
    }

    @GetMapping("/roles")
    public Set<Role> getListUserRole(@RequestParam String username) {
        return userService.getListUserRole(username);
    }
}

File: frontend/react-app/src/controller/UserController.ts
import {IUpdateDataUserInCabinet} from "../model/user/IUpdateDataUserInCabinet";
import api from "../http";
import {IUserDelivery} from "../model/user/IUserDelivery";

export class UserController {

    static async updateDataUserInCabinet(updateDataUserInCabinet:IUpdateDataUserInCabinet): Promise<IUpdateDataUserInCabinet> {
        return api.post<IUpdateDataUserInCabinet>("api/v1/user/data/update",updateDataUserInCabinet)
            .then(response => response.data);
    }

    static async changeDataUserAboutDelivery(userDeliveryUpdateData:IUserDelivery){
        return api.post<IUserDelivery>('api/v1/user/data/update/delivery',userDeliveryUpdateData);
    }


    static async getDataUserAboutDelivery():Promise<IUserDelivery> {
        return api.get<IUserDelivery>('api/v1/user/data/delivery').then(r=>r.data);
    }
}